chia/data_layer/data_store.py

```diff
@@ -930,49 +929,49 @@ async def get_proof_of_inclusion_by_key(
         self,
         key: bytes,
         tree_id: bytes32,
-        *,
-        lock: bool = True,
     ) -> ProofOfInclusion:
         """Collect the information for a proof of inclusion of a key and its value in
         the Merkle tree.
         """
-        async with self.db_wrapper.locked_transaction(lock=lock):
-            node = await self.get_node_by_key(key=key, tree_id=tree_id, lock=False)
-            return await self.get_proof_of_inclusion_by_hash(node_hash=node.hash, tree_id=tree_id, lock=False)
+        async with self.db_wrapper.read_db():
```

Comment:
it seems redundant to ask for a read connection object and then not use it. Is there some subtle reason to do this at all?